patches/server/0842-Added-EntityClimbEvent.patch

```diff
@@ -0,0 +1,44 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jan Trummer <jan.trummer@hotmail.com>
+Date: Mon, 3 Jan 2022 18:48:50 +0100
+Subject: [PATCH] Added EntityClimbEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index 43378561cf48f969f5bf1fd0db349415f4d1c866..25028d8e1880bff3ca251ebf3e09e812e5ae60a3 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -19,6 +19,7 @@ import java.util.Random;
+ import java.util.UUID;
+ import java.util.function.Predicate;
+ import javax.annotation.Nullable;
++
+ import net.minecraft.BlockUtil;
+ import net.minecraft.advancements.CriteriaTriggers;
+ import net.minecraft.commands.arguments.EntityAnchorArgument;
+@@ -139,7 +140,6 @@ import org.bukkit.event.entity.EntityTeleportEvent;
+ import org.bukkit.event.player.PlayerItemConsumeEvent;
+ // CraftBukkit end
+ 
+-import co.aikar.timings.MinecraftTimings; // Paper
+ 
+ public abstract class LivingEntity extends Entity {
+ 
+@@ -1849,6 +1849,12 @@ public abstract class LivingEntity extends Entity {
+             BlockState iblockdata = this.getFeetBlockState();
+ 
+             if (iblockdata.is((Tag) BlockTags.CLIMBABLE)) {
++                //Paper start
++                io.papermc.paper.event.entity.EntityClimbEvent event = null;
++                org.bukkit.craftbukkit.block.CraftBlock craftBlock = org.bukkit.craftbukkit.block.CraftBlock.at(this.level, blockposition);
++                event = new io.papermc.paper.event.entity.EntityClimbEvent(this.getBukkitLivingEntity(), craftBlock);
++                event.callEvent();
++                //Paper end
```

Comment:
If this is going to be a thing, the LivingEntity#onClimbable method is not the right place. CraftLivingEntity calls that when you call LivingEntity#isClimbing, and its called all over the place, in damage areas, collision checks, spigot's activation range. I feel like this event is going to fire seemingly randomly for plugins

Reply:
The event is only going to fire if the player is on block with the climbable tag, or do i miss something?

Reply:
right, but lets say the event fires, and in the event, something calls LivingEntity#isClimbing, its going to fire again, and then you've got an infinite loop right there.

Or lets say within the span of 1 tick multiple places call that method, and since its the same tick, the player is on the climbable block, so the event has fired twice, or 4 times in the same tick.

Reply:
That method, I think, is supposed to a check *if* the entity is on a climbable thing. If this event is going to be implemented, it should be where some action is taken based on the return value of onClimbable being true

Reply:
Okay thanks, i didnt think about that. 

I am gonna close this PR then, since the feedback has rather been that such a feature is not needed.